File: rust/src/map.rs
use std::collections::HashMap;
use std::hash::Hash;

pub struct CustomMap<T: Eq + Hash + Clone, V> {
    items: HashMap<T, V>,
}

impl<T: Eq + Hash + Clone, V> CustomMap<T, V> {
    pub fn new() -> CustomMap<T, V> {
        CustomMap {
            items: HashMap::new(),
        }
    }

    pub fn get(&self, key: T) -> Option<&V> {
        self.items.get(&key)
    }

    pub fn set(&mut self, key: T, value: V) {
        self.items.insert(key, value);
    }

    pub fn delete(&mut self, key: T) -> Option<V> {
        self.items.remove(&key)
    }

    pub fn size(&self) -> usize {
        self.items.len()
    }
}

File: java/src/Map.java
package src;
import java.util.HashMap;
import java.util.Map;

public class CustomMap<T, V> {
    private Map<T, V> items;

    public CustomMap() {
        this.items = new HashMap<>();
    }

    public V get(T key) {
        return items.get(key);
    }

    public void set(T key, V value) {
        items.put(key, value);
    }

    public V delete(T key) {
        V deletedValue = items.get(key);
        if (deletedValue != null) {
            items.remove(key);
            return deletedValue;
        }
        return null;
    }

    public int size() {
        return items.size();
    }
}
